src/OpenTelemetry.Api/DistributedContext/DistributedContext.cs

```diff
@@ -0,0 +1,66 @@
+﻿// <copyright file="DistributedContext.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenTelemetry.DistributedContext
+{
+    /// <summary>
+    /// Distributed context.
+    /// </summary>
+    public readonly struct DistributedContext
+    {
+        private readonly IEnumerable<DistributedContextEntry> entries;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DistributedContext"/> struct.
+        /// </summary>
+        /// <param name="entries">Entries for distributed context.</param>
+        public DistributedContext(IEnumerable<DistributedContextEntry> entries)
+        {
+            this.entries = entries;
+        }
+
+        /// <summary>
+        /// Gets the current <see cref="DistributedContext"/>.
+        /// </summary>
+        public static DistributedContext Current { get; /* TODO: to be implemented  */ }
```

Comment:
This being a `readonly struct` would this getter return a copy on each access? 
Just trying to understand what's the expectation here

Reply:
@tarekgh can you comment here?

Reply:
@bruno-garcia 

In regular cases it will make a copy. This will not be a concern as the struct size is small (currently it is IntPtr.Size). The compiler can optimize in other cases to avoid the copy (e.g. passing it as ```in``` parameter). This optimization doesn't happen if the struct is not readonly. The compiler can optimize in other cases when using its internal readonly fields too.

Reply:
Thanks for clarifying. I read when `in` came in in 7.x, nice to see in use.